Mathlib/Probability/Kernel/Composition/MeasureComp.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2025 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Lorenzo Luccioli
+-/
+import Mathlib.Probability.Kernel.Composition.MeasureCompProd
+
+/-!
+# Lemmas about the composition of a measure and a kernel
+
+This operation, for which we introduce the notation `∘ₘ`, takes `μ : Measure α` and
+`κ : Kernel α β` and creates `κ ∘ₘ μ : Measure β`. The integral of a function against `κ ∘ₘ μ` is
+`∫⁻ x, f x ∂(κ ∘ₘ μ) = ∫⁻ a, ∫⁻ b, f b ∂(κ a) ∂μ`.
+
+This file does not define composition but only introduces notation for
+`MeasureTheory.Measure.bind μ κ`.
+
+## Notations
+
+* `κ ∘ₘ μ = MeasureTheory.Measure.bind μ κ`, for `κ` a kernel.
+-/
+
+open scoped ENNReal
+
+open ProbabilityTheory MeasureTheory
+
+namespace MeasureTheory.Measure
+
+variable {α β γ : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β} {mγ : MeasurableSpace γ}
+  {μ ν : Measure α} {κ η : Kernel α β}
+
+/-- Composition of a measure and a kernel.
+
+Notation for `MeasureTheory.Measure.bind` -/
+scoped[ProbabilityTheory] notation3 κ " ∘ₘ " μ:100 => MeasureTheory.Measure.bind μ κ
+
+@[simp]
+lemma comp_apply_univ [IsMarkovKernel κ] : (κ ∘ₘ μ) Set.univ = μ Set.univ := by
+  simp [bind_apply .univ κ.measurable]
+
+lemma comp_assoc {η : Kernel β γ} : η ∘ₘ (κ ∘ₘ μ) = (η ∘ₖ κ) ∘ₘ μ :=
+  Measure.bind_bind κ.measurable η.measurable
+
+lemma deterministic_comp_eq_map {f : α → β} (hf : Measurable f) :
+    Kernel.deterministic f hf ∘ₘ μ = μ.map f :=
+  Measure.bind_dirac_eq_map μ hf
+
+@[simp]
+lemma id_comp : Kernel.id ∘ₘ μ = μ := by rw [Kernel.id, deterministic_comp_eq_map, Measure.map_id]
+
+lemma swap_comp {μ : Measure (α × β)} : (Kernel.swap α β) ∘ₘ μ = μ.map Prod.swap :=
+  deterministic_comp_eq_map measurable_swap
+
+@[simp]
+lemma const_comp {ν : Measure β} : (Kernel.const α ν) ∘ₘ μ = μ Set.univ • ν := μ.bind_const
+
+@[simp]
+lemma snd_compProd (μ : Measure α) [SFinite μ] (κ : Kernel α β) [IsSFiniteKernel κ] :
+    (μ ⊗ₘ κ).snd = κ ∘ₘ μ := by
+  ext s hs
+  rw [bind_apply hs κ.measurable, snd_apply hs, compProd_apply]
+  · rfl
+  · exact measurable_snd hs
+
+instance [SFinite μ] [IsSFiniteKernel κ] : SFinite (κ ∘ₘ μ) := by
+  rw [← snd_compProd]; infer_instance
+
+instance [IsFiniteMeasure μ] [IsFiniteKernel κ] : IsFiniteMeasure (κ ∘ₘ μ) := by
+  rw [← snd_compProd]; infer_instance
+
+instance [IsProbabilityMeasure μ] [IsMarkovKernel κ] : IsProbabilityMeasure (κ ∘ₘ μ) := by
```

Comment:
Can you add the corresponding instance for `[IsZeroOrProbabilityMeasure μ] [IsZeroOrMarkovKernel κ]`? Mayeb this means API simp lemmas for when the measure is zero, or the kernel is zero, but in both cases it looks like a good idea.